tasks/collect-data/collect-data.yaml

```diff
@@ -106,3 +107,41 @@ spec:
         DATA_PATH=$(params.subdirectory)/data.json
         echo -n $DATA_PATH > $(results.data.path)
         echo "$merged_output" > $(workspaces.data.path)/$DATA_PATH
+    - name: check-data-key-sources
+      image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+      script: |
+        #!/usr/bin/env sh
+        set -ex
+
+        DISALLOWED_KEYS_JSON='{
+            "Release": [
+                "releaseNotes.product_id",
+                "releaseNotes.cpe",
+                "releaseNotes.type"
+            ],
+            "ReleasePlan": [
+                "releaseNotes.product_id",
+                "releaseNotes.cpe",
+                "releaseNotes.type"
+            ],
+            "ReleasePlanAdmission": [
+            ]
+        }'
+
+        RC=0
+
+        check_source () { # Expected arguments are [CRD from DISALLOWED_KEYS_JSON, file]
+            for KEY in $(jq -r ".$1[]" <<< $DISALLOWED_KEYS_JSON) ; do
+                if [[ $(jq ".spec.data.$KEY" "$2") != "null" ]] ; then
```

Comment:
I was gonna suggest to use the `has` jq function for the check. But it seems it only works with an actual key name and not a subpath. So this works:

```
jq '.data | has("bla")' <<< '{"data": {"bla": {"bla": ""}}}
```

but this doesn't

```
jq '.data | has("bla.bla")' <<< '{"data": {"bla": {"bla": ""}}}
```

So what you have is good. Just sharing.